### PR TITLE
Updated validation logic to be explicit when modifying 

### PR DESCRIPTION
**Pull request by ghost**
## Describe your changes
As part of the validation there are times, per the standard, that a profile needs to be modified to conform to the spec. During the initial pass we had this logic in validate methods and did not explicitly state that the modification was occurring.

- renamed `validate_profile` to `conform_and_validate_profile` in order to convey the potential for change.
- renamed `validate_and_add_profile` to `conform_validate_and_add_profile` in order to convey the potential for change.
- added method `conform_validity_periods` to better show where the change is occurring.
- added comment as to why the conform is being performed.
- added logging to show the values that are being set.
- updated some methods to use `const`  on profiles where attributes are not being modified.

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [ ] If OCPP 2.0.1: I have updated the [OCPP 2.0.1 status document](https://github.com/EVerest/libocpp/tree/main/doc/ocpp_201_status.md)
- [x] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

### CODE REVIEW

File: lib/ocpp/v201/smart_charging.cpp [hunk 1]
```diff
@@ -675,9 +674,35 @@ bool SmartChargingHandler::is_overlapping_validity_period(const ChargingProfile&
     return std::any_of(charging_profiles.begin(), charging_profiles.end(), conflicts_with);
 }
 
+/// \brief sets attributes of the given \p charging_schedule_period according to the specification.
+/// 2.11. ChargingSchedulePeriodType if absetent numberPhases set to 3
```
Comment (gberardi-pillar) [inline lib/ocpp/v201/smart_charging.cpp]: s/absetent/absent/

File: lib/ocpp/v201/smart_charging.cpp [hunk 2]
```diff
@@ -675,9 +674,35 @@ bool SmartChargingHandler::is_overlapping_validity_period(const ChargingProfile&
...
-    profile.validTo = profile.validTo.value_or(ocpp::DateTime(date::utc_clock::time_point::max()));
+    if (!profile.validFrom.has_value()) {
+        auto validFrom = ocpp::DateTime();
+        EVLOG_debug << "Conforming profile: " << profile.id << " added validFrom as " << validFrom;
+        profile.validFrom = validFrom;
+    }
+
+    if (!profile.validTo.has_value()) {
+        auto validTo = ocpp::DateTime(date::utc_clock::time_point::max());
+        EVLOG_debug << "Conforming profile: " << profile.id << " added validFrom as " << validTo;
```
Comment (gberardi-pillar) [inline lib/ocpp/v201/smart_charging.cpp]: s/added validFrom/added validTo/

**Review by gberardi-pillar — APPROVED** [review]
Other than the typos noted above, LGTM.

**Review by folkengine — APPROVED** [review]
What Franco said.

**Pull request merged**